src/backend/expungeservice/expunger/expunger.py

```diff
@@ -123,4 +126,46 @@ def _class_b_felonies(charges):
         for charge in charges:
             if isinstance(charge, FelonyClassB):
                 class_b_felonies.append(charge)
-        return class_b_felonies
\ No newline at end of file
+        return class_b_felonies
+
+    @staticmethod
+    def _mark_missing_or_unknown_dispositions_as_record_errors(record):
+        cases_with_missing_disposition : List[str] = []
+        cases_with_unknown_disposition : List[str] = []
+        for charge in record.charges:
+            if not charge.skip_analysis():
+                case_number = charge.case()().case_number
+                if not charge.disposition:
+                    if not case_number in cases_with_missing_disposition:
+                        cases_with_missing_disposition.append(case_number)
+                elif charge.disposition.status == DispositionStatus.UNKNOWN:
+                    if not case_number in cases_with_unknown_disposition:
+                        cases_with_unknown_disposition.append(case_number)
+            message_shared_content = (
+                "This is likely an error in the OECI database. " +
+                "Time analysis is ignoring these charges and may be inaccurate for other charges.")
+            if cases_with_missing_disposition:
```

Comment:
Will this ever happen if `charge.skip_analysis()` is `True`?

Reply:
Not sure what you mean. The result of `skip_analysis()` is always unaffected by the disposition, so the charges with `skip_analysis() == True` never should and never do get collected here.

Reply:
I was just asking a question. Now that I know the answer, I would suggest to indent this but that should naturally happen if you extract a method to handle each non-skipped charge.